database/migrations/2017_12_29_152706_create_templates_table.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Lara\Club;
+use Lara\Schedule;
+use Lara\Section;
+use Lara\Shift;
+
+class CreateTemplatesTable extends Migration
+{
+    const BD_TEMPLATE_NAME = 'BD Template';
+
+    const BD_SECTION_NAME = 'bd-Club';
+
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table("shifts", function(Blueprint $table) {
+           $table->integer('schedule_id')->unsigned()->nullable()->change();
+        });
+        Schema::create('templates', function (Blueprint $table) {
+            $table->increments('id');
+            $table->timestamps();
+
+            $table->string('title');
+            $table->string('subtitle');
+            $table->smallInteger('type');
+            $table->unsignedBigInteger('section_id');
+            $table->time('time_preparation_start');
+            $table->time('time_start');
+            $table->time('time_end');
+            $table->longText('public_info');
+            $table->longText('private_details');
+            $table->boolean('is_private');
+            $table->float("price_tickets_normal")->nullable();
+            $table->float("price_tickets_external")->nullable();
+            $table->float("price_normal")->nullable();
+            $table->float("price_external")->nullable();
+            $table->boolean('facebook_needed')->default(false);
+        });
+
+        Schema::create('shift_template', function (Blueprint $table) {
+            $table->integer('template_id')->unsigned()->index();
+            $table->integer('shift_id')->unsigned()->index();
+
+            $table->foreign('template_id')->references('id')->on('templates')->onDelete('cascade');
+            $table->foreign('shift_id')->references('id')->on('shifts')->onDelete('cascade');
+        });
+
+        Schema::create('section_template', function (Blueprint $table) {
+            $table->integer('template_id')->unsigned()->index();
+            $table->integer('section_id')->unsigned()->index();
+
+            $table->foreign('template_id')->references('id')->on('templates')->onDelete('cascade');
+        });
+
+        Schema::table("club_events", function(Blueprint $table) {
+            $table->integer('template_id')->unsigned()->nullable();
+            $table->foreign('template_id')->references('id')->on('templates');
+        });
+
```

Comment:
the bd template is special because it is created programmaticaly
so there is no real event as a template, just the schedule, because it was not needed
this requires a special handling